c7n/resources/awslambda.py

```diff
@@ -255,6 +255,68 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'KMSKeyArn'
 
 
+@AWSLambda.filter_registry.register('check-xray-tracing-enabled')
+class LambdaCheckXrayTracingEnabled(ValueFilter):
+    """Filters lambda functions with xray tracing not enabled
+
+    :example:
+
+    .. code-block:: yaml
+
+            filters:
+              - type: check-xray-tracing-enabled
+
+    """
+
+    schema = type_schema('check-xray-tracing-enabled')
+    permissions = ('lambda:ListFunctions',)
+
+    def process(self, resources, event=None):
+        results = []
+        for resource in resources:
+            if resource["TracingConfig"]["Mode"] != "Active":
+                results.append(resource)
+        return results
+
+
+@AWSLambda.action_registry.register('enable-xray-tracing')
+class LambdaEnableXrayTracing(Action):
+    """
+        This action allows for enable Xray tracing to Active
+       :example:
+       .. code-block:: yaml
+           actions:
+             - type: enable-xray-tracing
+    """
+
+    schema = type_schema('enable-xray-tracing')
+    permissions = ("lambda:UpdateFunctionConfiguration",)
+
+    def process(self, resources):
+        """
+            Enables the Xray Tracing for the function.
+
+            Args:
+                resources: AWS lamdba resources
+            Returns:
+                None
+        """
+        client = local_session(self.manager.session_factory).client('lambda')
+
+        for resource in resources:
+            function_name = resource["FunctionName"]
+            self.log.info(f"Enabling Xray tracing for lambda {function_name}")
+            try:
+                client.update_function_configuration(
```

Comment:
we should check if tracing is already enabled and if it is skip

Reply:
Good catch! updated code.